src/main/java/John.java

```diff
@@ -0,0 +1,118 @@
+            }
+            //Mark Command
+            else if (input.startsWith("unmark") || input.startsWith("Unmark")){
+                try {
+                    StringValidator.validateUnmarkFormat(input);
+                    mark(input, tasks, false);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+
+            else if (input.startsWith("mark") || input.startsWith("Mark")){
+                try {
+                    StringValidator.validateMarkFormat(input);
+                    mark(input, tasks, true);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+            //commands.Todo Command
+            else if (input.startsWith("todo") || input.startsWith("commands.Todo")) {
+                try {
+                    StringValidator.validateTodoFormat(input);
+                    Todo todo = new Todo(Todo.parse(input));
+                    todo.addTo(tasks);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+
+            //commands.Deadline Command
+            else if (input.startsWith("deadline") || input.startsWith("commands.Deadline")) {
+                try {
+                    StringValidator.validateDeadlineFormat(input);
+                    Deadline deadline = new Deadline(Deadline.parseName(input), Deadline.parseDate(input));
+                    deadline.addTo(tasks);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+
+            //commands.Event Command
+            else if (input.startsWith("event") || input.startsWith("commands.Event")) {
+                try {
+                    StringValidator.validateEventFormat(input);
+                    Event event = new Event(Event.parseName(input), Event.parseStart(input), Event.parseEnd(input));
+                    event.addTo(tasks);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+
+            //Unknown Command
+            else {
+                System.out.println("Unknown Command: " + input);
+            }
+        }
+    }
+
+    public static void mark(String command, List<Task> list, Boolean b) {
+        int taskNum = Integer.parseInt(command.substring(command.indexOf(" ")+1));
+        if (taskNum <= list.size() && taskNum != 0){
+            list.get(taskNum-1).setDone(b);
+            if (b) {
+                System.out.println("Nice! I've marked this task as done:\n" + list.get(taskNum-1));
+            }
+            else {
+                System.out.println("OK, I've marked this task as not done yet:\n" + list.get(taskNum-1));
+            }
+        }
+        else{
```

Comment:
maybe include a space after else, since all the other else statements have one before "{"